### DISCUSSION TITLE
load html created by nicegui

### DISCUSSION THREAD

**Discussion by retsyo** [webui-dev/webui#248] [Show and tell]
[nicegui](https://nicegui.io/) is a web-based gui for python, which uses Tailwind and Quasar. We do not need to write html/javascript. The appilcation can be opened in web-brower or, like [webui](https://github.com/webui-dev), in a native window. 

I am using [nim webui](https://github.com/webui-dev/nim-webui).

# first step, successful
When this python code
```python
from nicegui import ui

ui.label('Hello NiceGUI!').style('color: #FF0000; font-weight: bold')

ui.run()
```
opens a browser, I view the source code and copy it to `index.html` 

Then I make a nim code to use [nim webui](https://github.com/webui-dev/nim-webui)

```nim
import webui

let s = readFile("index.html")

let window = newWindow() # Create a new Window
window.show(s) # Show the window with html content

wait() # Wait until the window gets closed
```

Last, I copy everything under `nicegui\static\` to `_nicegui\1.3.16\static\`

Nice, the compiled nim code runs as expected

![nicegui2webui](https://github.com/webui-dev/webui/assets/7960913/bbb49cfe-4ac6-430f-a8bf-83357e771899)

# second try, failed
for a more complex python code
```python
from nicegui import ui

ui.icon('thumb_up')
ui.markdown('This is **Markdown**.')
ui.html('This is <strong>HTML</strong>.')
with ui.row():
    ui.label('CSS').style('color: #888; font-weight: bold')
    ui.label('Tailwind').classes('font-serif')
    ui.label('Quasar').classes('q-ml-xl')
ui.link('NiceGUI on GitHub', 'https://github.com/zauberzeug/nicegui')

ui.run()
```
nothing appears on webui :(
![nicegui2webui_failed](https://github.com/webui-dev/webui/assets/7960913/c26b8ea7-234c-4b0a-8c46-b87581b5a3dc)

**Comment by fibodevy** [comment]
How does your `index.html` generated with `nicegui` looks like? Any errors in devtools console?

**Comment by hassandraga** [comment]
We may need more details, so it would be beneficial if you could try again using the [debug version](https://github.com/webui-dev/webui/releases/tag/nightly) of WebUI and share the logs with us.

**Reply by hassandraga** [reply]
1. Download the archive for your platform
2. Extract the archive and copy the WebUI library from the `debug` folder
3. Replace the WebUI lib file in the nicegui project with the debug one
4. Run the nicegui software from the terminal and copy all the logs

**Comment by retsyo** [comment]
finnally it is truns out that 
1. `successful.html` can be rendered in an external webbrowser, so `load_successful_html.nim` does render correctly
2. `failed.html` can't be rendered in an external webbrowser, and says "Connection lost. Trying to reconnect...". So `load_failed_html.nim` does not render the web page(or I do something wrong?)

In a word, the problem is not in webui, but in the HTML produced by nicegui.
[webui_nimlang_load_html.zip](https://github.com/webui-dev/webui/files/12981139/webui_nimlang_load_html.zip)

**Comment by retsyo** [comment]
Currently, it seems that we have to design `HTML` file for webui by writing `Javascript` in`HTML`.  But I do expect that we can program in just one language, just like [nicegui](https://nicegui.io/) does.
